usaspending_api/etl/management/commands/load_transactions_in_delta.py

```diff
@@ -0,0 +1,416 @@
+                        transaction_id_lookup.published_fabs_id IS NOT NULL
+                        AND NOT EXISTS (
+                            SELECT *
+                            FROM raw.published_fabs
+                            WHERE transaction_id_lookup.published_fabs_id = published_fabs.published_fabs_id
+                        )
+                    )
+            """
+        else:
+            id_col = "id" if self.etl_level == "transaction_normalized" else "transaction_id"
+            subquery = f"""
+                SELECT {id_col} AS id_to_remove
+                FROM int.{self.etl_level}
+                WHERE
+                    NOT EXISTS (
+                        SELECT *
+                        FROM int.transaction_id_lookup
+                        WHERE transaction_id_lookup.id = {self.etl_level}.{id_col}
+                    )
+            """
+
+        sql = f"""
+            MERGE INTO int.{self.etl_level}
+            USING (
+                {subquery}
+            ) AS deleted_records
+            ON {self.etl_level}.{id_col} = deleted_records.id_to_remove
+            WHEN MATCHED
+            THEN DELETE
+        """
+        return sql
+
+    def initial_run(self):
+        """
+        TODO:
+            - Still need to create migration that creates the sequence in Postgres
+            - External Data values need to be added for all etl_levels (including transaction_ids)
+            - Script to backfill all IDs in the lookup table
+        """
+        self.spark.sql("USE int")
+
+        # This CREATE TABLE statement should be moved out of this; left in place for now during local development
+        spark_s3_bucket = CONFIG.SPARK_S3_BUCKET
+        delta_lake_s3_bucket = CONFIG.DELTA_LAKE_S3_PATH
+        destination_database = "int"
+        destination_table = "transaction_id_lookup"
+        self.spark.sql(
+            f"""
+            CREATE TABLE IF NOT EXISTS {destination_table} (
+                id LONG NOT NULL,
+                detached_award_procurement_id INTEGER,
+                published_fabs_id INTEGER,
+                transaction_unique_id STRING NOT NULL
+            )
+            USING DELTA
+            LOCATION 's3a://{spark_s3_bucket}/{delta_lake_s3_bucket}/{destination_database}/{destination_table}'
+        """
+        )
+
+        # Before running INSERT, make sure expected keys have no NULLs
+        table_names = ("int.transaction_normalized", "raw.detached_award_procurement", "raw.published_fabs")
+        key_names = ("transaction_unique_id", "detached_award_proc_unique", "afa_generated_unique")
+        for table_name, key_name in zip(table_names, key_names):
+            test_results = self.spark.sql(
+                f"SELECT COUNT(*) AS count FROM {table_name} WHERE {key_name} IS NULL").collect()
+
+            assert test_results[0]['count'] == 0, \
+                f"Found {test_results[0]['count']} NULLs in {key_name} in table {table_name}!"
```

Comment:
`assert` works sometimes, but there's always the risk that where this python code runs, assertions don't raise errors. So best to raise an error here instead.